src/yaml.cpp

```diff
@@ -147,21 +147,31 @@ YamlResult<void> mergeYamlMaps(YAML::Node & map_a, YAML::Node map_b) {
 }
 
 // Uitility function to check whether a value is present in a vector.
-bool Contains(const std::vector<int> &list, int x)
-{
+bool Contains(const std::vector<int> &list, int x){
 	return std::find(list.begin(), list.end(), x) != list.end();
 }
 
+// Utility function to check whether key in map_b is present in map_a.
+bool orderedDictFind(YAML::Node map_a, YAML::Node map_b, std::vector<int> & checked, int index){
```

Comment:
This function should only take a single ordered dict and search for a string key. It much to specialized now.

Reply:
It's not searching for a string key now. It just checks if a certain index has the string key.

I would really like you to follow the function signatures that I proposed earlier:
```c++
    std::optional<std::size_t> orderedDictFind(YAML::Node parent, std::string const & name);

    void orderedDictUpdateOrInsert(YAML::Node parent, std::string const & name, YAML::Node contents);
```